components/prism-lawtex.js

```diff
@@ -0,0 +1,32 @@
+Prism.languages.lawtex = {
+  'comment': [
+    {
+      pattern: /(^|[^\\])\/\*[\s\S]*?(?:\*\/|$)/,
+      lookbehind: true,
+      greedy: true
+    },
+    {
+      pattern: /(^|[^\\:])\/\/.*/,
+      lookbehind: true,
+      greedy: true
+    }
+  ],
+  'string': {
+    pattern: /(^|[^\\](?:\\\\)*)(")(?:\\[\s\S]|\$\([^)]+\)|\$(?!\()|`[^`]+`|(?!\2)[^\\`$])*\2/,
+    lookbehind: true,
+    greedy: true,
+    inside: {
+      'escape': /\\./
+    }
+  },
+  'keyword': /\b(?:metainfo|head|body|foot|extra|where|extends|externs|template|list|struct|dependency|branch|topic|loop|period|if|else|elseif|foreach|print|use)\b/,
+  'important': /\b(?:this|value|id|mask|transient|itemsOnPage|language|style|type|default|description|tags|statement|overview|components|audience|inputs|functionalities|warnings|upgrades|title|name|label|order|mandatory|version|index|lower|upper|separator|request|element|help|tip|atomic|fields|loaders|options|key|operation|declarations|operations)\b/,
+  'logical': /\b(?:NOT|AND|OR|XOR|EXISTS|FORALL|IN)/,
+  'class-name': /\b(?:List|Object|String|Integer|Real|Boolean|Date|Time|Text|Document|Currency|Vector|Struct)\b/,
+  'function': /\b[A-Z].*?\b/,    
+  'boolean': /\b(?:true|false|null)\b/,
+  'operator': /(?:\[|\]|==|~=|<=|>=|<|>|\|)/,
+  'assignment': /=/,
+  'arithmetic': /(?:\-|\+|\&|\*|\/|%|\^)/,
+  'number': /(?:(?:0(?:x|X)[0-9a-fA-F]*)|(?:\+|-)?\b(?:(?:[0-9]+(?:\.[0-9]*)?)|(?:\.[0-9]+))(?:(?:e|E)(?:\+|-)?[0-9]+)?)(?:[LlFfUuDd]|UL|ul)?\b/
```

Comment:
Changes:
1. `0x` probably isn't a valid number.
2. I added a `\b` before `0x`.
3. Made the whole regex ignore case.
4. Some other simplifications.

```suggestion
  'number': /(?:\b0x[0-9A-F]+|[+-]?\b(?:\d+(?:\.\d*)?|\.\d+)(?:E[+-]?\d+)?)(?:[LFUD]|UL)?\b/i
```

Reply:
The sintax highlighter used by the language (programmed into a Sublime package) considers `0x` a valid number even if it is appended to the end of a word (not fulfilling the `\b` requirement); however, I cannot think of a situation in which this would be used. So I would not be against the simplifications you have proposed.